code-challenges.test.js

```diff
@@ -10,42 +11,197 @@
+
+ // hung up on the result in the .push
+
+// Expected output: ["Ford Prefect is a hitchhiker.", "Zaphod Beeblebrox is president of the galaxy.", "Arthur Dent is a radio employee."]
+
+
+//-----------broken for now
+
+
+// declare a function called jobs
+// use array.map to check array
+// that upcases the first index of the first and last name
+// that takes the name and occupation index 
+// from const people and puts them together in a string interpolation
+// make it pass the test 
+
+// const jobs = (array) => {
+//   return array.map(firstletter => {
+//      firstletter.name.split(" ").map(value => value.toUpperCase(value[0],value[0].toUpperCase()))
+//       let jobpeps = array.join(" ")
+//       let nameTitle = `${jobpeps} is a ${occupation}.`
+//       return nameTitle
+//   })
+// }
+
+
+// Test Suites: 1 failed, 1 total
+// Tests:       1 failed, 1 total
+
+// 1 failed, 1 total
+// Test Suites: 1 failed, 1 total
+// Tests:       1 failed, 1 total
+
+// getting hung up on occupation
+
+
+
+
+
 
 // --------------------2) Create a function that takes in a mixed data array and returns an array of only the REMAINDERS of the numbers when divided by 3.
 
 // a) Create a test with an expect statement using the variables provided.
 
-const hodgepodge1 = [23, "Heyyyy!", 45, -10, 0, "Yo", false]
-// Expected output: [ 2, 0, -1, 0 ]
-const hodgepodge2 = [5, "Hola", 43, -34, "greetings", true]
-// Expected output: [ 2, 1, -1 ]
+//a describe method that lists the name of the function OR naming of the particular test.
+describe("mix", () => {
+
+  // a test/it method, nested within the describe block, that in plain words, describes that the function does.
+  it("that takes in a mixed data array and returns an array of only the REMAINDERS of the numbers when divided by 3.", () => {
+
+    const hodgepodge1 = [23, "Heyyyy!", 45, -10, 0, "Yo", false]
 
+    const hodgepodge2 = [5, "Hola", 43, -34, "greetings", true]
+
+    //an expect method, nested within the test block, calling on the hello() function, followed by the .toEqual() matcher that checks the expected output of the function return.
+    expect(mix(hodgepodge1)).toEqual([ 2, 0, -1, 0 ])
+    expect(mix(hodgepodge2)).toEqual([ 2, 1, -1 ])
+  })
+})
 
 // b) Create the function that makes the test pass.
 
+// take in a mixed array
+// use filter to take out everything that is not a number
+// use .map to take the array with new values and get the desired math 
+// return an array with the REMAINDERS of the left over numbers
+// make it pass the test
+
+
+
+const mix = (array) => {
+    return array.filter(value => 
+    typeof (value) === "number").map(value => value % 3)
+
+}
+
+// Test Suites: 1 passed, 1 total
+// Tests:       1 passed, 1 total
+
+// Test Suites: 1 passed, 1 total
+// Tests:       1 passed, 1 total
+
+// Test Suites: 1 failed, 1 total
+// Tests:       1 failed, 1 total
+
+// Expected output: [ 2, 0, -1, 0 ]
+// Expected output: [ 2, 1, -1 ]
+
```

Comment:
Nice work. Very clean and concise.

Reply:
Thank you the first 2 question were the hardest.